src/apprt/gtk/Surface.zig

```diff
@@ -2502,3 +2502,35 @@ fn gtkStreamError(media_file: *gtk.MediaFile, _: *gobject.ParamSpec, _: ?*anyopa
 fn gtkStreamEnded(media_file: *gtk.MediaFile, _: *gobject.ParamSpec, _: ?*anyopaque) callconv(.c) void {
     media_file.unref();
 }
+
+pub fn showChildExited(self: *Surface, info: apprt.surface.Message.ChildExited) (error{})!bool {
+    if (!adw_version.supportsBanner()) return false;
+
+    const warning_box = gtk.Box.new(.vertical, 0);
+
+    warning_box.as(gtk.Widget).setHalign(.fill);
+    warning_box.as(gtk.Widget).setValign(.end);
+
+    const warning_text = if (info.exit_code == 0)
+        i18n._("⚠️ Process exited normally. Press any key to close the terminal.")
+    else
+        i18n._("⚠️ Process exited abnormally. Press any key to close the terminal.");
+
+    const banner = adw.Banner.new(warning_text);
+    banner.setRevealed(1);
+
+    const banner_widget = banner.as(gtk.Widget);
+
+    banner_widget.addCssClass("child_exited");
+
+    if (info.exit_code == 0)
+        banner_widget.addCssClass("child_exited_normally")
+    else
+        banner_widget.addCssClass("child_exited_abnormally");
```

Comment:
Honestly I probably would just do the dumb route and have two copies of the banner as blueprints, and just hardcode the style and the label text